### ISSUE TITLE
AWS: Wily cannot install Docker (upstream issue)

### ISSUE THREAD

**Issue by justinsb** [kubernetes/kubernetes#21853]
AWS Wily cannot install Docker, due to: https://github.com/docker/docker/issues/18444

**Comment by justinsb** [comment]
Also vivid.

Here's a portion of /var/log/syslog showing the installation failure:


```
Feb 24 04:10:50 ip-172-20-0-9 rc.local[640]: [INFO    ] Running state [/var/cache/docker-install/docker-engine_1.9.1-0~vivid_amd64.deb] at time 04:10:50.238664
Feb 24 04:10:50 ip-172-20-0-9 rc.local[640]: [INFO    ] Executing state file.managed for /var/cache/docker-install/docker-engine_1.9.1-0~vivid_amd64.deb
Feb 24 04:10:50 ip-172-20-0-9 rc.local[640]: [INFO    ] Starting new HTTP connection (1): apt.dockerproject.org
Feb 24 04:10:50 ip-172-20-0-9 rc.local[640]: [INFO    ] File changed:
Feb 24 04:10:50 ip-172-20-0-9 rc.local[640]: New file
Feb 24 04:10:50 ip-172-20-0-9 rc.local[640]: [INFO    ] Completed state [/var/cache/docker-install/docker-engine_1.9.1-0~vivid_amd64.deb] at time 04:10:50.471226
Feb 24 04:10:50 ip-172-20-0-9 rc.local[640]: [INFO    ] Running state [docker-upgrade] at time 04:10:50.472118
Feb 24 04:10:50 ip-172-20-0-9 rc.local[640]: [INFO    ] Executing state pkg.installed for docker-upgrade
Feb 24 04:10:51 ip-172-20-0-9 rc.local[640]: [INFO    ] Executing command ['dpkg', '-i', '--force-confold', '/var/cache/docker-install/docker-engine_1.9.1-0~vivid_
amd64.deb'] in directory '/root'
Feb 24 04:10:54 ip-172-20-0-9 systemd[1]: Reloading.
Feb 24 04:10:55 ip-172-20-0-9 systemd[1]: message repeated 3 times: [ Reloading.]
Feb 24 04:10:55 ip-172-20-0-9 systemd[1]: Starting Docker Application Container Engine...
Feb 24 04:10:55 ip-172-20-0-9 docker[20700]: time="2016-02-24T04:10:55.270055996Z" level=fatal msg="No sockets found"
Feb 24 04:10:55 ip-172-20-0-9 systemd[1]: docker.service: main process exited, code=exited, status=1/FAILURE
Feb 24 04:10:55 ip-172-20-0-9 systemd[1]: Failed to start Docker Application Container Engine.
Feb 24 04:10:55 ip-172-20-0-9 systemd[1]: Unit docker.service entered failed state.
Feb 24 04:10:55 ip-172-20-0-9 systemd[1]: docker.service failed.
Feb 24 04:10:55 ip-172-20-0-9 systemd[1]: Reloading.
Feb 24 04:10:56 ip-172-20-0-9 rc.local[640]: [ERROR   ] Command ['dpkg', '-i', '--force-confold', '/var/cache/docker-install/docker-engine_1.9.1-0~vivid_amd64.deb'
] failed with return code: 1
Feb 24 04:10:56 ip-172-20-0-9 rc.local[640]: [ERROR   ] output: Selecting previously unselected package docker-engine.
Feb 24 04:10:56 ip-172-20-0-9 rc.local[640]: (Reading database ... 87057 files and directories currently installed.)
Feb 24 04:10:56 ip-172-20-0-9 rc.local[640]: Preparing to unpack .../docker-engine_1.9.1-0~vivid_amd64.deb ...
Feb 24 04:10:56 ip-172-20-0-9 rc.local[640]: Unpacking docker-engine (1.9.1-0~vivid) ...
Feb 24 04:10:56 ip-172-20-0-9 rc.local[640]: Setting up docker-engine (1.9.1-0~vivid) ...
Feb 24 04:10:56 ip-172-20-0-9 rc.local[640]: Job for docker.service failed. See "systemctl status docker.service" and "journalctl -xe" for details.
Feb 24 04:10:56 ip-172-20-0-9 rc.local[640]: invoke-rc.d: initscript docker, action "start" failed.
Feb 24 04:10:56 ip-172-20-0-9 rc.local[640]: dpkg: error processing package docker-engine (--install):
Feb 24 04:10:56 ip-172-20-0-9 rc.local[640]: subprocess installed post-installation script returned error exit status 1
Feb 24 04:10:56 ip-172-20-0-9 rc.local[640]: Processing triggers for ureadahead (0.100.0-19) ...
Feb 24 04:10:56 ip-172-20-0-9 rc.local[640]: Processing triggers for systemd (219-7ubuntu6) ...
Feb 24 04:10:56 ip-172-20-0-9 rc.local[640]: Processing triggers for man-db (2.7.0.2-5) ...
Feb 24 04:10:56 ip-172-20-0-9 rc.local[640]: Errors were encountered while processing:
Feb 24 04:10:56 ip-172-20-0-9 rc.local[640]: docker-engine
```

Salt then fails to install anything that depends on docker (e.g. kube-apiserver)

**Comment by justinsb** [comment]
One potential workaround might be to add a policy to disable start:

`/usr/sbin/policy-rc.d:`

```
#!/bin/sh
exit 101
```

(must be executable)

**Comment by justinsb** [comment]
(Working on this; it's pretty bad...  I'm trying policy-rc.d.  Creating a bash script to install the package, so we can be sure that we cleanup policy-rc.d.  Wishing I was allowed to use go here :-) )

**Comment by justinsb** [comment]
#21904 will fix this

**Comment by justinsb** [comment]
#21904 fixed this

**Comment by trodriguez-photobox** [comment]
Does it mean that the [official method](http://kubernetes.io/docs/getting-started-guides/aws/) for installing Kubernetes on AWS doesn't work for v1.2.0?

**Comment by justinsb** [comment]
@trodriguez-photobox why do you say that?  This was an issue, and it was closed because it was fixed as part of development of 1.2.

**Comment by trodriguez-photobox** [comment]
It was just a question as I was experiencing a really similar behaviour because of #23633
